### PR TITLE
fix: Keyboard persistence bug fixed

### LINKED ISSUE #752: Keyboard persists during 3rd party transfer
**Summary:**
Under 3rd party transfer when the remarks are entered and the review is selected , the keyboard doesn't get dismissed. 

**Steps to reproduce:**
Go to 3rd party transfer and add the remarks and select REVIEW.

**Screenshot**
device-2018-03-18-144442

**Expected behaviour:**
The keyboard should be dismissed.

### PR DESCRIPTION
**Pull request by sridharjajoo**
Fixes #752 

device-2018-03-18-231838

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by sridharjajoo** [comment]
@therajanmaurya @dilpreet96 Kindly review!

File: app/src/main/java/org/mifos/mobilebanking/utils/Utils.java [hunk 1]
```diff
@@ -83,4 +85,12 @@ public static String formatTransactionType(String type) {
         }
         return builder.toString();
     }
+
+    public static void hideKeyboard(Activity activity) {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/utils/Utils.java]: @sridharjajoo Please create this method in `BaseActivity` and use from there.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/base/BaseActivity.java [hunk 2]
```diff
@@ -252,4 +254,12 @@ public int stackCount() {
         return getSupportFragmentManager().getBackStackEntryCount();
     }
 
+
+    public static void hideKeyboard(Activity activity) {
+        InputMethodManager inputMethodManager =
+                (InputMethodManager) activity.getSystemService(
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/activities/base/BaseActivity.java]: Please use Context instead of Activity

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/base/BaseActivity.java (hunk 2 above)
Comment (sridharjajoo) [reply app/src/main/java/org/mifos/mobilebanking/ui/activities/base/BaseActivity.java]: Actually Activity is used for using the getCurrentFocus( ) . What I can do is pass context as the parameter and convert it to activity something like this:

 public static void hideKeyboard(Context context) {
      
  Activity activity = (Activity) context;

        InputMethodManager inputMethodManager =
                (InputMethodManager) context.getSystemService(
                        Activity.INPUT_METHOD_SERVICE);

        inputMethodManager.hideSoftInputFromWindow(
                activity.getCurrentFocus().getWindowToken(), 0);
    }

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**